lib/input/reader/amqp_1.go

```diff
@@ -193,14 +246,137 @@ func (a *AMQP1) ReadWithContext(ctx context.Context) (types.Message, AsyncAckFn,
 
 	msg.Append(part)
 
+	var done chan struct{}
+	if a.conf.RenewLock {
+		done = a.startRenewJob(amqpMsg)
+	}
+
 	return msg, func(ctx context.Context, res types.Response) error {
+		if done != nil {
+			close(done)
+		}
+
 		if res.Error() != nil {
 			return amqpMsg.Modify(ctx, true, false, amqpMsg.Annotations)
 		}
 		return amqpMsg.Accept(ctx)
 	}, nil
 }
 
+func (a *AMQP1) startRenewJob(amqpMsg *amqp.Message) chan struct{} {
+	done := make(chan struct{})
```

Comment:
Where do we release the channel in case of error?

Reply:
Channel is closed by ack function that is called despite if there is an error (in such case noAck is sent) or success.

Reply:
Maybe we need to clean it when leaving the go routine to not depend on the Ack.

Reply:
But if we close it here, how would we know that it's closed in ack to not close it a second time?

Reply:
Maybe we could use a https://golang.org/pkg/context/#WithDeadline instead and not deal with channel?

Reply:
But then we will have a specific time to process the message and we would continue to renew lock even after sending ack what isn't correct.

Reply:
Ok, I see.  Then maybe we can use a cancellable context instead of the done channel and keep the timer?